samples/minimum_criteria.rst

```diff
@@ -0,0 +1,98 @@
+.. _definition_and_minimum_criteria:
+
+Sample Definition and Minimum Criteria
+######################################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem,
+or module. For example, Kernel services, drivers, protocols, etc.
+Samples should be limited in scope and should focus only on demonstrating non-trivial or
+essential aspects of the subsystem(s) or module(s) being highlighted.
+
+Minimum Sample Criteria
+=======================
+
+1. Samples must not be used as a testcase.
+++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added.
+    Testcases are mandatory, and having samples may be optional.
+  * The primary purpose of a sample is to provide a reference to the user.
+  * Samples must not use Zephyr's testing framework.
+
+    * Must not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+    * Must not use zasserts or asserts in samples.
```

Comment:
Not in agreement with this, asserts have a lot of use and preventing them from being used in samples just hides potential issues that would otherwise be very obvious.

Reply:
Asserts is what I mean. zasserts for tests only, but normal asserts should be allowed in samples

Reply:
Changed this.